includes/promoted-jobs/class-wp-job-manager-promoted-jobs-api.php

```diff
@@ -147,4 +172,31 @@ public function update_job_status( $request ) {
 			200
 		);
 	}
+
+	/**
+	 * Get the job data, but only if the user has permission to manage the job and the token is valid.
+	 *
+	 * @param WP_REST_Request $request Full data about the request.
+	 * @return WP_REST_Response|WP_Error The response, or WP_Error on failure.
+	 */
+	public function get_job_data( $request ) {
+		$token    = $request->get_param( 'token' );
+		$user_id  = $request->get_param( 'user_id' );
+		$job_id   = $request->get_param( 'job_id' );
+		$verified = WP_Job_Manager_Site_Trust_Token::instance()->validate( 'user', $user_id, $token );
+		$result   = [
+			'verified' => $verified,
+		];
+		if ( $verified ) {
+			// We only want to return the job data if the specified user has permission to manage the job.
```

Comment:
Sorry for identifying it later, but I think it would make more sense as part of the `permission_callback`, no? Anyway, I think it would be just a different way to handle it, so feel free to skip it for now if you don't consider it important.

Reply:
I think this enters the same discussion [that we had on my other PR](https://github.com/Automattic/WP-Job-Manager/pull/2489#discussion_r1256648213): If we return 403 (which I think will happen if `permission_callback` returns `false`), it will be harder (maybe impossible?) to identify whether the request failed because the token verification failed, or because something else failed. WDYT?

That other PR was merged back into this one, so this code is appearing again here. Sorry for that confusion too.

Reply:
Ah, I'm sorry we had discussed this one before!
I remember that one, but I thought it was another part!